Update rerun-failed-tests-only.adoc (#8117)
circleci/circleci-docs: jekyll/_cci2/rerun-failed-tests-only.adoc

* Jobs that are older than the xref:persist-data#custom-storage-usage[retention period] for workspaces for the organization cannot be re-run with "Re-run failed tests only".
* Jobs that upload code coverage reports link:https://discuss.circleci.com/t/product-launch-re-run-failed-tests-only-circleci-tests-run/47775/3?u=sebastian-lerner[may see issues during a re-run].

[#troubleshooting]
== Troubleshooting

> If after configuring `circleci tests run`, you are seeing *all tests* be re-run after clicking "Rerun failed tests only":

Reviewer: ```suggestion
After configuring `circleci tests run`, if you see *all tests* are re-run after clicking "Rerun failed tests only":
```

1. Ensure that the `--verbose` setting is enabled when invoking `circleci tests run`. This will display what "tests" `circleci tests run` is receiving on a "re-run".

> 2. Use `store_artifacts` to upload to CircleCI the JUnit XML that contains test results.  This is the same file(s) that is being uploaded to CircleCI with `store_test_results`

Reviewer: ```suggestion
2. Use xref:configuration-reference#storeartifacts[`store_artifacts`] to upload  the JUnit XML that contains test results to CircleCI.  This is the same file(s) that is being uploaded to CircleCI with `store_test_results`
```

> 3. Manually inspect the newly uplaoded JUnit XML via the Artifacts tab and ensure that there is a `file=` attribute or a `classname` attribute.  If neither of those are present, you will see unexpected behavior when trying to re-run.  Follow the instructions on this page to ensure that the test runner you are using is outputting its JUnit XML test results with a `file` (preferred) or `classname` attribute.  Comment in our link:https://discuss.circleci.com/t/product-launch-re-run-failed-tests-only-circleci-tests-run/47775/48[community forum] if you are still stuck.

Reviewer: ```suggestion
3. Manually inspect the newly uplaoded JUnit XML via the **Artifacts** tab and ensure that there is a `file=` attribute or a `classname` attribute.  If neither are present, you will see unexpected behavior when trying to re-run.  Follow the instructions on this page to ensure that the test runner you are using is outputting its JUnit XML test results with a `file` (preferred) or `classname` attribute.  Comment in our link:https://discuss.circleci.com/t/product-launch-re-run-failed-tests-only-circleci-tests-run/47775/48[community forum] if you are still stuck.
```
